packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/FishjamClientInternal.kt

```diff
@@ -90,8 +94,23 @@ internal class FishjamClientInternal(
         ?.firstOrNull { track -> track.getRTCEngineId() == trackId }
 
   fun connect(config: Config) {
+    this.config = config
+    this.reconnectionManager =
+      ReconnectionManager(config.reconnectConfig) {
+        reconnect(config)
+      }
     peerConnectionManager.addListener(this)
     rtcEngineCommunication.addListener(this)
+    reconnectionManager.addListener(listener)
+    setupWebsocket(config)
+  }
+
+  private fun reconnect(config: Config) {
+    recreateTracks()
+    setupWebsocket(config)
+  }
+
+  private fun setupWebsocket(config: Config) {
```

Comment:
nit: in most places in code we use `webSocket`
```suggestion
  private fun setupWebSocket(config: Config) {
```